app/src/main/java/org/ethereum/dappstore/MainActivity.kt

```diff
@@ -12,19 +13,26 @@ import androidx.navigation.compose.NavHost
 import androidx.navigation.compose.composable
 import androidx.navigation.compose.currentBackStackEntryAsState
 import androidx.navigation.compose.rememberNavController
+import kotlinx.coroutines.*
 import org.ethereum.dappstore.ui.Screen
 import org.ethereum.dappstore.ui.components.*
 import org.ethereum.dappstore.ui.theme.DAppStoreTheme
 import org.ethereum.dappstore.ui.theme.Selected
 import org.ethereum.dappstore.ui.theme.Unselected
+import org.ethereum.dappstore.logic.AppUpdater
 
 class MainActivity : FragmentActivity() {
 
     private val bottomNavScreens =
         listOf(Screen.MainNav.Home, Screen.MainNav.User, Screen.MainNav.Settings)
-
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
+        val context: Context = this;
+        runBlocking {
```

Comment:
runBlocking can block the main thread and is not properly scoped. You should use `lifecycleScope.launchWhenStarted(Dispatchers.default) {}` instead. You might need to import `androidx.lifecycle:lifecycle-runtime-ktx:2.2.0` to get the nice kotlin extension.

Reply:
Ah, didn't know it would run it on the main-thread. It was the first thing that I found on running suspending functions 😅, fixed it now